source/slang-capture-replay/slang-session.cpp

```diff
@@ -0,0 +1,196 @@
+        return pTypeLayoutReflection;
+    }
+
+    SLANG_NO_THROW slang::TypeReflection* SessionCapture::getContainerType(
+        slang::TypeReflection* elementType,
+        slang::ContainerType containerType,
+        ISlangBlob** outDiagnostics)
+    {
+        slangCaptureLog(LogLevel::Verbose, "%s\n", "SessionCapture::getContainerType");
+        slang::TypeReflection* pTypeReflection = m_actualSession->getContainerType(elementType, containerType, outDiagnostics);
+        return pTypeReflection;
+    }
+
+    SLANG_NO_THROW slang::TypeReflection* SessionCapture::getDynamicType()
+    {
+        slangCaptureLog(LogLevel::Verbose, "%s\n", "SessionCapture::getDynamicType");
+        slang::TypeReflection* pTypeReflection = m_actualSession->getDynamicType();
+        return pTypeReflection;
+    }
+
+    SLANG_NO_THROW SlangResult SessionCapture::getTypeRTTIMangledName(
+        slang::TypeReflection* type,
+        ISlangBlob** outNameBlob)
+    {
+        slangCaptureLog(LogLevel::Verbose, "%s\n", "SessionCapture::getTypeRTTIMangledName");
+        SlangResult result = m_actualSession->getTypeRTTIMangledName(type, outNameBlob);
+        return result;
+    }
+
+    SLANG_NO_THROW SlangResult SessionCapture::getTypeConformanceWitnessMangledName(
+        slang::TypeReflection* type,
+        slang::TypeReflection* interfaceType,
+        ISlangBlob** outNameBlob)
+    {
+        slangCaptureLog(LogLevel::Verbose, "%s\n", "SessionCapture::getTypeConformanceWitnessMangledName");
+        SlangResult result = m_actualSession->getTypeConformanceWitnessMangledName(type, interfaceType, outNameBlob);
+        return result;
+    }
+
+    SLANG_NO_THROW SlangResult SessionCapture::getTypeConformanceWitnessSequentialID(
+        slang::TypeReflection* type,
+        slang::TypeReflection* interfaceType,
+        uint32_t*              outId)
+    {
+        slangCaptureLog(LogLevel::Verbose, "%s\n", "SessionCapture::getTypeConformanceWitnessSequentialID");
+        SlangResult result = m_actualSession->getTypeConformanceWitnessSequentialID(type, interfaceType, outId);
+        return result;
+    }
+
+    SLANG_NO_THROW SlangResult SessionCapture::createTypeConformanceComponentType(
+        slang::TypeReflection* type,
+        slang::TypeReflection* interfaceType,
+        slang::ITypeConformance** outConformance,
+        SlangInt conformanceIdOverride,
+        ISlangBlob** outDiagnostics)
+    {
+        slangCaptureLog(LogLevel::Verbose, "%s\n", "SessionCapture::createTypeConformanceComponentType");
+        SlangResult result = m_actualSession->createTypeConformanceComponentType(type, interfaceType, outConformance, conformanceIdOverride, outDiagnostics);
+        return result;
+    }
+
+    SLANG_NO_THROW SlangResult SessionCapture::createCompileRequest(
+        SlangCompileRequest**   outCompileRequest)
+    {
+        slangCaptureLog(LogLevel::Verbose, "%s\n", "SessionCapture::createCompileRequest");
```

Comment:
You could do without the indirection,
```
slangCaptureLog(LogLevel::Verbose, "SessionCapture::createCompileRequest\n");
```
Or you can avoid any mistakes on the function names by using the `__FUNC__` macro.
```
slangCaptureLog(LogLevel::Verbose, "%s:%d:%s\n", __FILE__, __LINE__, __FUNC__);
```

Reply:
Yes, we should be making use of `__FUNC__`.

Reply:
OK, this log call is actually generated by copilot, I will update that.